docs/samples/graph/README.md

```diff
@@ -0,0 +1,251 @@
+splitter-model-00001-deployment-c5ccc95d5-lxhnb                   2/2     Running   0          15m
+sklearn-iris-predictor-default-00001-deployment-8495cbf8cbdqfjg   2/2     Running   0          15m
+xgboost-iris-predictor-default-00001-deployment-7b86bcdcf-7njrl   2/2     Running   0          15m
+
+kubectl get isvc
+NAME              URL                                                    READY   PREV   LATEST   PREVROLLEDOUTREVISION   LATESTREADYREVISION                       AGE
+sklearn-iris      http://sklearn-iris.default.10.166.15.29.sslip.io      True           100                              sklearn-iris-predictor-default-00001      80m
+xgboost-iris      http://xgboost-iris.default.10.166.15.29.sslip.io      True           100                              xgboost-iris-predictor-default-00001      80m
+
+kubectl get ig
+NAME            URL                                                  READY   AGE
+splitter-model   http://splitter-model.default.10.166.15.29.sslip.io   True    15m
+```
+3. Tesing `graph`.
+```shell
+curl http://splitter-model.default.10.166.15.29.sslip.io -d @./iris-input.json
+``` 
+***Expect result***
+```shell
+{"treeModel":{"predictions":[1,1]}}
+```
+[***Demo yaml***](splitter.yaml)
+### **3.4 [Ensemble](ensemble.yaml)**
+***Test steps***
+
+1. Deploy the demo `isvc` and `graph` 
+```shell 
+kubectl apply -f switch.yaml
+```
+2. Waiting for `isvc` and `graph` up.
+```shell
+kubectl get pods
+NAME                                                              READY   STATUS    RESTARTS   AGE
+ensemble-model-00001-deployment-7d48f984b6-qqqsh                  2/2     Running   0          32s
+sklearn-iris-predictor-default-00001-deployment-8495cbf8cbdqfjg   2/2     Running   0          52s
+xgboost-iris-predictor-default-00001-deployment-7b86bcdcf-7njrl   2/2     Running   0          50s
+
+kubectl get isvc
+NAME              URL                                                    READY   PREV   LATEST   PREVROLLEDOUTREVISION   LATESTREADYREVISION                       AGE
+sklearn-iris      http://sklearn-iris.default.10.166.15.29.sslip.io      True           100                              sklearn-iris-predictor-default-00001      80m
+xgboost-iris      http://xgboost-iris.default.10.166.15.29.sslip.io      True           100                              xgboost-iris-predictor-default-00001      80m
+
+kubectl get ig
+NAME            URL                                                  READY   AGE
+ensemble-model   http://ensemble-model.default.10.166.15.29.sslip.io   True    15m
+```
+3. Tesing `graph`.
+```shell
+curl http://ensemble-model.default.10.166.15.29.sslip.io -d @./iris-input.json
+``` 
+***Expect result***
+```shell
+{"http://sklearn-iris.default.svc.cluster.local/v2/models/sklearn-iris/infer":{"predictions":[1,1]},"http://xgboost-iris.default.svc.cluster.local/v2/models/xgboost-iris/infer":{"predictions":[1,1]}}
```

Comment:
Why keys for predictions are complete service URLs ? Why not just use the service name ike `sklearn-iris` and `xgboost-iris` as the key ?
Won't output be more readable like that ? 
```
{
  "sklearn-iris": {
    "predictions": [
      1,
      1
    ]
  },
  "xgboost-iris": {
    "predictions": [
      1,
      1
    ]
  }
}
```

Reply:
Think about the scenario when ensemble’s output is passed onto downstream model in the graph. Then working on the json with URL as keys is not going to a great developer/user experience.

Reply:
though service name is optional and in case only service url is provided we won't be able to populate that, but agreed we can at least optimize for the service name case.

Reply:
I have added a new field `stepName` for this, so the merged response will be populated with `stepName` as the key instead of using `serviceUrl`.

Reply:
`stepName` field would work fine too.